core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifier.java

```diff
@@ -98,9 +98,9 @@ void notifyCacheEntryPassivated(K key, V value, boolean pre,
     */
    void notifyTransactionRegistered(GlobalTransaction globalTransaction, boolean isOriginLocal);
 
-   void notifyDataRehashed(ConsistentHash oldCH, ConsistentHash newCH, ConsistentHash unionCH, int newTopologyId, boolean pre);
+   void notifyDataRehashed(ConsistentHash oldCH, ConsistentHash newCH, ConsistentHash unionCH, int newTopologyId, boolean pre, CacheTopology.Phase phase);
```

Comment:
Good point about exposing too much to the user. Originally, before my solution introduced a new `CacheTopology.Phase,` the `StateConsumer` interacted directly with the `ConflictManager` and we decided it was best to use the Notifier. But obviously I now need to share additional information between the components. I will refactor so that the `StateConsumer` interacts directly and remove the Phase from the notifier events as before.